src/acme/ui/CompleteATicketUI.java

```diff
@@ -0,0 +1,162 @@
+    lblCourier.setBounds(30, 87, 83, 16);
+    add(lblCourier);
+    
+    JLabel lblActPackageId = new JLabel(this.ticketToComplete.getId() + "");
+    lblActPackageId.setBounds(125, 59, 150, 16);
+    add(lblActPackageId);
+    
+    JLabel lblActCourier = new JLabel(this.ticketToComplete.getCourier().getName() + "");
+    lblActCourier.setBounds(125, 87, 150, 16);
+    add(lblActCourier);
+    //--------------------------------------------------------------
+    
+    
+    //times section-------------------------------------------------
+    JLabel lblActualDepartureTime = new JLabel("Actual Departure time");
+    lblActualDepartureTime.setBounds(30, 155, 145, 16);
+    add(lblActualDepartureTime);
+    
+    this.dateTimePickerDeparture = new DateTimePicker();
+    this.dateTimePickerDeparture.setBounds(200, 155, 400, 30);
+    add(this.dateTimePickerDeparture);
+    
+    JLabel lblActualPickUpTime = new JLabel("Actual Pick Up time");
+    lblActualPickUpTime.setBounds(30, 205, 135, 16);
+    add(lblActualPickUpTime);
+    
+    this.dateTimePickerPickup = new DateTimePicker();
+    this.dateTimePickerPickup.setBounds(200, 205, 400, 30);
+    add(this.dateTimePickerPickup);
+    
+    JLabel lblActualDeliveryTime = new JLabel("Actual Delivery time");
+    lblActualDeliveryTime.setBounds(30, 255, 145, 16);
+    add(lblActualDeliveryTime);
+    
+    this.dateTimePickerDelivery = new DateTimePicker();
+    this.dateTimePickerDelivery.setBounds(200, 255, 400, 30);
+    add(this.dateTimePickerDelivery);
+    //-------------------------------------------------------------
+    
+    //cancel and save buttons 
+    JButton btnCancel = new JButton("Cancel");
+    btnCancel.setBounds(364, 305, 117, 29);
+    btnCancel.setBackground(lightBlue);
+    btnCancel.setOpaque(true);
+    btnCancel.addActionListener(new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
+        clickedCancel();
+      }
+    });
+    add(btnCancel);
+    
+    JButton btnSave = new JButton("Save");
+    btnSave.setBounds(480, 305, 117, 29);
+    btnSave.setBackground(lightBlue);
+    btnSave.setOpaque(true);
+    btnSave.addActionListener(new ActionListener() {
+      public void actionPerformed(ActionEvent e) {
+        clickedSave();
+      }
+    });
+    add(btnSave);
+   
+    this.revalidate();
+    this.repaint();
+  }
+  
+  //this method checks to make sure the times make sense in the basic sense of things and then 
+  //completes the ticket
+  public void clickedSave()
+  {
+    if(this.dateTimePickerDeparture.getDateTimeStrict() != null && this.dateTimePickerPickup.getDateTimeStrict() != null
+        && this.dateTimePickerDelivery.getDateTimeStrict() != null)
+    {
+      if(this.dateTimePickerPickup.getDateTimeStrict().isBefore(this.dateTimePickerDeparture.getDateTimeStrict()) &&
+          this.dateTimePickerDelivery.getDateTimeStrict().isBefore(this.dateTimePickerPickup.getDateTimeStrict()))
```

Comment:
Shouldn't pickup come before delivery?

Reply:
Yeah, I got that sections backwards. I fixed it now.